### PR TITLE
feature: Noise operators

### PR DESCRIPTION
**Pull request by xiaosi-xu**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-schemas-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by speller26 — COMMENTED** [review]
You'll need to run `tox` to fix the formatting issues

**Comment by xiaosi-xu** [comment]
> You'll need to run `tox` to fix the formatting issues

Is there an instruction for that?

**Comment by speller26** [comment]
Yep, just run

```
tox
```

And commit the resulting changes.

File: src/braket/ir/jaqcd/instructions.py [hunk 1]
```diff
@@ -696,3 +703,199 @@ class Type(str, Enum):
...
+    """
+    Bit Flip noise channel.
+
+    Attributes:
+        type (str): The instruction type. default = "bit_flip". (type) is
+            optional. This should be unique among all instruction types.
+        target (int): The target qubit. This is an int >= 0.
+
+    Examples:
+        >>> Bit_Flip(target=1, probability=0.1)
```
Comment (speller26) [inline src/braket/ir/jaqcd/instructions.py:718]: no underscore

File: src/braket/ir/jaqcd/instructions.py [hunk 2]
```diff
@@ -696,3 +703,199 @@ class Type(str, Enum):
...
+    """
+    Phase Flip noise channel.
+
+    Attributes:
+        type (str): The instruction type. default = "phase_flip". (type) is
+            optional. This should be unique among all instruction types.
+        target (int): The target qubit. This is an int >= 0.
+
+    Examples:
+        >>> Phase_Flip(target=1, probability=0.1)
```
Comment (speller26) [inline src/braket/ir/jaqcd/instructions.py:737]: no underscore

File: src/braket/ir/jaqcd/shared_models.py [hunk 3]
```diff
@@ -129,12 +129,113 @@ class Angle(BaseModel):
...
+            NaN is not an allowable input.
+
+    Examples:
+        >>> SingleProbability_34(probability=0.5)
+    """
+
+    probability: confloat(ge=float("0.0"), le=float("0.75"))
+
+
+class SingleProbability_1516(BaseModel):
```
Comment (speller26) [inline src/braket/ir/jaqcd/shared_models.py:162]: No underscores needed in the class names, also the class names should be descriptive; `_34` and `_1516` aren't useful. Also, why isn't `SingleProbability` sufficient?

File: src/braket/ir/jaqcd/instructions.py (hunk 1 above)
Comment (xiaosi-xu) [reply src/braket/ir/jaqcd/instructions.py]: Thanks

File: src/braket/ir/jaqcd/shared_models.py (hunk 3 above)
Comment (xiaosi-xu) [reply src/braket/ir/jaqcd/shared_models.py:162]: No, Quest has the probability limit for some noise channels, like [0,0.5] for bit_flip and [0,0.75] for depolarizing channel, and we decided to apply that for this launch, so _34 is for channels with limit [0,3/4] and _1516 is for [0,15/16].

Comment (speller26) [reply src/braket/ir/jaqcd/shared_models.py:162]: Worth documenting the 3/4 and 15/16 limits on the angles, and explaining why they exist.

Comment (xiaosi-xu) [reply src/braket/ir/jaqcd/shared_models.py]: sure.

**Review by virajchau — APPROVED** [review]

**Review by mdpratt — APPROVED** [review]

**Pull request merged**